### ISSUE TITLE
NULL pointer dereference when using ESAPI with no TCTI

### ISSUE THREAD

**Issue by niooss-ledger** [tpm2-software/tpm2-pytss#230]
Hello,
With current master (https://github.com/tpm2-software/tpm2-pytss/commits/8f9faf2b843e3ce7f8ded07456acf458f4f3890f), the following code triggers a NULL pointer dereference (seen as a NULL pointer dereference):

```python
from tpm2_pytss import ESAPI
esapi = ESAPI()
print(esapi.tcti)
```

Using `gdb` to get a stack trace leads to:

```text
(gdb) bt
#0  read_raw_unsigned_data (size=4, target=0x8 <error: Cannot access memory at address 0x8>) at c/_cffi_backend.c:926
#1  convert_to_object (data=0x8 <error: Cannot access memory at address 0x8>, ct=0x7fee84323d40) at c/_cffi_backend.c:1107
#2  0x0000000000512536 in _PyEval_EvalFrameDefault (tstate=<optimized out>, f=<optimized out>, throwflag=<optimized out>) at ../Python/ceval.c:2994
#3  0x00000000005291c3 in _PyEval_EvalFrame (throwflag=0, 
    f=Frame 0x7fee842f1040, for file /usr/local/lib/python3.9/dist-packages/tpm2_pytss/TCTI.py, line 13, in __init__ (self=<TCTI(_v1=<_cffi_backend._CDataBase at remote 0x7fee8244a2a0>) at remote 0x7fee843be0a0>, ctx=<_cffi_backend._CDataBase at remote 0x7fee8438aed0>), tstate=0xffcbb0)
    at ../Include/internal/pycore_ceval.h:40
```

`TCTI.py, line 13` is https://github.com/tpm2-software/tpm2-pytss/blob/8f9faf2b843e3ce7f8ded07456acf458f4f3890f/tpm2_pytss/TCTI.py#L10-L13

When debugging the issue, it appears that the `ctx` argument is `NULL`. More precisely here are the results when reproducing the calls in https://github.com/tpm2-software/tpm2-pytss/blob/8f9faf2b843e3ce7f8ded07456acf458f4f3890f/tpm2_pytss/ESAPI.py#L153-L155

```python
>>> from tpm2_pytss import ESAPI
>>> esapi = ESAPI()
>>>
>>> import tpm2_pytss.types
>>> tctx = tpm2_pytss.types.ffi.new("TSS2_TCTI_CONTEXT **")
>>> tpm2_pytss.types.lib.Esys_GetTcti(esapi._ctx, tctx)
0
>>> tctx
<cdata 'struct TSS2_TCTI_OPAQUE_CONTEXT_BLOB * *' owning 8 bytes>
>>> tctx[0]
<cdata 'struct TSS2_TCTI_OPAQUE_CONTEXT_BLOB *' NULL>
```
 
In short, `Esys_GetTcti` returns `NULL`, `self._v1 = ffi.cast("TSS2_TCTI_CONTEXT_COMMON_V1 *", ctx)` casts a `NULL` pointer and `if self._v1.version == 2` dereferences this pointer.

I guess this issue can be fixed by supporting constructing `TCTI` objets with `NULL` pointers, or by detecting such an error in `ESAPI.get_tcti`.

By the way, I reproduced this issue in a clean container running Debian sid with libtss2 3.1.0-3, and swtpm running on TCP port 2321.
